packages/turbo/src/BridgeComponent.ts

```diff
@@ -0,0 +1,89 @@
+import { Component } from 'react';
+import type { EmitterSubscription } from 'react-native';
+import type {
+  VisitableViewModule,
+  StradaEvent,
+  StradaMessages,
+  StradaReactComponentProps,
+} from './types';
+import { getNativeModule, registerMessageEventListener } from './common';
+
+const RNVisitableViewModule = getNativeModule<VisitableViewModule>(
+  'RNVisitableViewModule'
+);
+
+class BridgeComponent extends Component {
+  private name: string;
+  private url: string;
+  private sessionHandle: string;
+  private eventHandlerEventSubscription?: EmitterSubscription;
+  private previousMessages: StradaMessages = {};
+
+  constructor(props: StradaReactComponentProps) {
+    super(props);
+    this.url = props.url;
+    this.name = props.name;
+    this.sessionHandle = props.sessionHandle;
+    this.didReceive = this.didReceive.bind(this);
+    this.onReceive = this.onReceive.bind(this);
+    this.replyTo = this.replyTo.bind(this);
+  }
+
+  componentDidMount() {
+    this.eventHandlerEventSubscription = registerMessageEventListener(
+      this.name,
+      this.didReceive
+    );
+  }
+
+  componentWillUnmount() {
+    this.eventHandlerEventSubscription?.remove();
+  }
+
+  didReceive(e: object) {
```

Comment:
It's a tiny bit confusing to have both `didReceive` and `onReceive`.

I also think that it doesn't make sense to have `didReceive` to be defined for each BridgeComponent; I think it'd be better if the components only received messages that are relevant for their instance (ie. where the URL matches). The BridgeComponents should not have to care about ignoring messages meant for other instances than itself, it feels kinda leaky.

Reply:
Maybe add a helper, eg. `registerStradaMessageEventListener(component)` that sets up the event listener with URL filtering alá:

```js
function registerStradaMessageEventListener(component) {
   const listener = registerMessageEventListener(component.name, (e: object) => {
     const message = e as StradaEvent;
     if (message.data.metadata.url !== component.url) { return }
  
     component.onReceive(message)
   })
}